frontend/src/utils/fetch-with-timeout.ts

```diff
@@ -0,0 +1,55 @@
+import { Dispatch } from 'redux';
+import { addNotification } from '../context/notificationStateSlice';
+
+interface FetchWithTimeoutOptions extends RequestInit {
+  timeout?: number;
+}
+
+export const ANALYSIS_REQUEST_TIMEOUT = 60000;
+
+const DEFAULT_REQUEST_TIMEOUT = 15000;
+
+export const fetchWithTimeout = async (
+  resource: RequestInfo,
+  dispatch: Dispatch,
+  options?: FetchWithTimeoutOptions,
+  fetchErrorMessage?: string,
```

Comment:
Why covering the real error message with a custom one ?
I would have just propagate the real error message of the HTTP response.

Reply:
The underlying error is not always very telling to non-developers / users

Reply:
Because the HTTP error message most of the times is not very informative friendly so what I did is to give the option to pass a better errorMessage than the HTTP's one. Which will involve the url that we are hitting and is going to give some more infformation about the actual problem. Also if we have a generic error handler there for network and abort messages, will give us a better handling to handle all other local errors in  higher order functions, when we throw them.

Reply:
@HarryMytilinaios maybe we could still log the actual error in the console? wdyt?

Reply:
of course